eZ/Bundle/EzPublishCoreBundle/Resources/views/content_fields.html.twig

```diff
@@ -357,55 +357,59 @@
+    {% set latitude = latitude|default(0) %}
+    {% set longitude = longitude|default(0) %}
         <script>
             if (typeof(window.ezgmaplocationMapsScriptLoaded) == 'undefined') {
                 (function (win, doc) {
                     var myScript = document.createElement('script');
-                    myScript.src = '//maps.googleapis.com/maps/api/js{% if apiKey %}?key={{ apiKey }}{% endif %}';
+                    var myCss = document.createElement('link');
+                    myScript.src = 'https://unpkg.com/leaflet@1.3.1/dist/leaflet.js';
+                    myScript.crossorigin = "";
+                    myCss.rel = "stylesheet";
+                    myCss.href = "https://unpkg.com/leaflet@1.3.1/dist/leaflet.css";
+                    myCss.crossorigin = "";
                     win.ezgmaplocationMapsScriptLoaded = true;
+                    doc.body.appendChild(myCss);
                     doc.body.appendChild(myScript);
                 })(window, document)
             }
         </script>
         <script type="text/javascript">
-        (function(win, doc) {
-            var mapView = function (mapId, latitude, longitude) {
-                var mapElt = doc.getElementById("{{ mapId }}"),
-                    startPoint = new google.maps.LatLng(latitude, longitude),
-                    zoom = {{ zoom }},
-                    draggable = {{ draggable }};
-
-                new google.maps.Marker({
-                    map: new google.maps.Map(mapElt, {
-                        center: startPoint,
-                        zoom: zoom,
-                        draggable: draggable,
-                        scrollwheel: {{ scrollWheel }},
-                        mapTypeId: google.maps.MapTypeId.{{ mapType }}
-                    }),
-                    position: startPoint
-                });
-            }
-            if ( win.addEventListener ) {
-                win.addEventListener(
-                    'load',
-                    function () {
-                        mapView("{{ mapId }}", {{ latitude }}, {{ longitude }});
-                    },
-                    false
-                );
-            } else if ( win.attachEvent ) {
-                win.attachEvent(
-                    'onload',
-                    function () {
-                        mapView("{{ mapId }}", {{ latitude }}, {{ longitude }});
-                    }
-                );
-            }
-        })(window, document);
+            (function (win) {
+                var mapView = function (mapId, latitude, longitude) {
+                    var mapConfig = {
+                        dragging: {{ zoom }},
+                        scrollWheelZoom: {{ scrollWheel }}
+                    };
+                    var map = L.map(mapId, mapConfig).setView([latitude, longitude], {{ zoom }});
```

Comment:
You don't have to call method `.setView()` it can be set in config:
```js
...
zoom: {{ zoom }},
center: [latitude, longitude],
...
```

and coords can be chached in variable because are used second time in `L.marker()`